### ISSUE TITLE
pls improve performance on very long file

### ISSUE THREAD

**Issue by leewi9** [typora/typora-issues#68]
Now i'm writing a md file which is about 121200 words, and when typing it just becomes very lag.

Also it takes almost several minutes to open the file every time.

**Comment by DanBennettUK** [comment]
Possibly related to #58 :)

**Comment by DanBennettUK** [comment]
Just an update on this issue - still stands in latest version.

It's actually painful to type now. Every word takes a second to appear. Having to give up until fixed :-(.

(I'll test this on OS X at some point with the same file)

**Comment by abnerlee** [comment]
Currently typora cannot handle very large file well, we will improve this in future

**Comment by leewi9** [comment]
If so could you define a max word/line/size count per file that won't hurt performance, estimation also ok.


2016年3月4日 +0800 17:37 AbnerLee<notifications@github.com>，写道：

**Comment by DanBennettUK** [comment]
Just to add - this is also a problem on OS X. So not just a Windows thing. 55762 Word file hangs on both.

Must be the rendering? Don't know if Typora is ever going to manage to work with such a large file if it's constantly rendering the markdown :cry:

**Comment by abnerlee** [comment]
The performance should have been largely improved in new version.

Please reopen this issue if anyone found performance issue

**Comment by DanBennettUK** [comment]
Sadly I'm finding performance is just as bad (at least on the Windows version) on a large file. Unable to even type at the moment :(. Had to give up using it for now

**Comment by abnerlee** [comment]
@DanBennettUK did you use v0.9.0? Which size of the file did you writing?

**Comment by abnerlee** [comment]
OK, I tested, opening speed is improved on Windows, but writing still has laps. So I will re-open this to solve improve writing speed on Windows.

But in OS X, the performance is basically acceptable.

**Comment by DanBennettUK** [comment]
442KB File here. 70,000+ words.

It opens faster but it takes ages until I can scroll to the end of the file to continue it. Then typing is just... painful.

Glad to hear about OS X. I'll try test the same file there at some point but sadly I can't use OS X at work :-1: Boss is anti-OS X!

**Comment by leewi9** [comment]
woo I'll definitely quit if got a boss like this ;)


2016年4月5日 +0800 00:16 Dan Bennett<notifications@github.com>，写道：

**Comment by Fogest** [comment]
On my more powerful gaming/coding machine I don't see issues with performance on typing, however when I am using my ultrabook for note taking in class it is fairly laggy when typing. There is delays for characters appearing. I assume this is because on every keystroke it is re generating the display, however on an ultrabook style system with less power this approach doesn't seem to work too well.

I am not sure how it is currently being done, however it seems like some kind of partial rendering needs to be done on larger documents to keep peak performance. For now I can work with creating a new markdown file each lecture to avoid longer files, however it would be more convenient having one bigger file for each of my classes.

**Comment by geeeeeeeeek** [comment]
The performance is just as bad in macOS. Especially when you are typing Chinese characters with an IME, and it would take > 1s delay for the character to be inserted into the document. Totally unacceptable. When you are typing English letters, it just looks a bit laggy.

My document is 70k words long for now.

**Comment by lafengnan** [comment]
The performance is bad as content length growing.  My document is only 7k words, including Chinese characters and math formula,  by now.

Typora version is 0.9.9.9.4.2, 
OS: macOS Sierra 10.12.5
CPU: Intel Core i5 2.3GHz
RAM: 10GB 1333 MHz DDR3
Disk: Toshiba SSD

**Comment by xuhaomin0430** [comment]
Same problem occurred. The performance becomes worse as the words grow up, including Chinese characters and formula. Specifically, the response speed turns out slow(1-2 seconds lagging) when typing Chinese Characters and math formulas.

Typora version: 0.9.9.11.2
OS: macOS Sierra 10.12.6
Processor:3.1 GHz Intel Core i5
RAM: 8 GB 2133 MHz LPDDR3

**Comment by shellcodesniper** [comment]
Same as xuhaomin0430 so quote that.. but i'm korean so when i use korean chracters it shows slow down but english is okay
Same problem occurred. The performance becomes worse as the words grow up, including **Korean** characters and formula. Specifically, the response speed turns out slow(1-2 seconds lagging) when typing **Korean** Characters.

Typora version: 0.9.9.11.2
OS: macOS Sierra 10.13.2
Processor:2.7 GHz Intel Core i5
RAM: 8 GB 2133 MHz LPDDR3

**Comment by necan** [comment]
still not fixed.

**Comment by genericyzh** [comment]
#1800

**Comment by vassudanagunta** [comment]
I've been experiencing this in the past month or so. I'm not sure if it's because my docs are now longer and more complex, of its because Typora has gotten worse in this area. 

@abnerlee have you confirmed this yourself? Do you need sample files or memory dumps when this is happening or are you good?

**Comment by abnerlee** [comment]
@vassudanagunta If possible, sample files would be helpful. You could send to us <hi@typora.io>

**Comment by lloyd094** [comment]
I can confirm the issue still exists - my file is ~22,700 words, but is rather table and blockquote heavy. I'll send a sample file to your email with the issue number listed.

**Comment by zendai** [comment]
Same here with the latest version installed (Windows). Once the file gets beyond 100-150k, you can feel it gets sluggish. Also, to load the file (I use the directory on the left side which makes it very comfortable to switch between files) it takes 7-8 secs, which is very slow and kills the idea of switching at all. Tried to split up my work into multiple small files, but then the TOC is messed up. I'm using lot of code snippets, besides that standard formatting (headings, bold). Typora is the best Markdown editor I've used (I tried tons), but this slowness makes me move away from the editor. I'll rather have a less nice two panel version (editor and preview) that is responsive than a great looking one which is not. I'll try VS Code now. If you can further increase the responsiveness with large files, I think you got the best product on the market for MarkDown.

**Comment by syockit** [comment]
I'm wondering if Typora queues some events periodically that builds when not handled? Everytime I put my PC to sleep and resume the next day, the document it has in view always hang e.g. not responding to scroll or cursor movement, while the rest of the GUI like the menu bar still works (but anything that acts on the document seems to have not apparent effect).

This is on version 0.9.70.

**Comment by Petelin** [comment]
Hi, i am a coder write blog with typora. 
i meet this issue to, when there is a file with a lot of words 
https://github.com/Petelin/luckblog/issues/41 it is very slow to scroll 

This is the only thing typora is not the best. But it also the only reason i cann't use typora.

**Comment by peetteerr** [comment]
Same question for math rendering(It is going to be extremely slow when having inserted more than ten math blocks ). Maybe following festure can be of some help.https://github.com/typora/typora-issues/issues/499#issuecomment-578359465

**Comment by oliverproud** [comment]
@abnerlee to answer your question from #3251, my markdown file contains no images, just 300 words, but is in the parent directory which contains many other files and folders, of which have sub-folders.

**Comment by HelperInCa** [comment]
when there are 200+ pictures(each one is less than 1MB) in a file, it will be lagged and sometimes stuck. 
- 351 lines, 9127characters
- Typora Version 0.9.9.32.1 (4191)
- macOS 10.15.2
- MacBook Pro 16''(i7 16G)

**Comment by truefalsename** [comment]
Hey, I am also observing some lag in a file with a couple of small pictures but loads of math formulas. It takes ~10 seconds or so for typora to open and close a math box.  I also notice some lag when typing text on a new line (i type it in and it shows up after a few seconds).

I am on the latest windows 10 and the latest typora on a MS surface pro 6.

**Comment by HelperInCa** [comment]
> when there are 200+ pictures(each one is less than 1MB) in a file, it will be lagged and sometimes stuck. #3463
> 
> * 351 lines, 9127characters
> * Typora Version 0.9.9.32.1 (4191)
> * macOS 10.15.2
> * MacBook Pro 16''(i7 16G)

I have to use the Visual Studio Code to open that file, which there is no lag.

**Comment by ccshao** [comment]
In linux typora 0.9.89, typora performance is much better.

**Comment by fabiensc0ville** [comment]
I also want to state in both _Dell XPS 9570_, and the latest _MacBook Pro 16' i9, the lagginess makes Typora impossible to work with for more than 100 000 characters and **text-only** documents.

As @HelperInCa mentioned though, it works fine on VSCode and SublimeText.

As it is 4 years old issue, I am wondering if it can be solved or if the software would have to be rewritten from scratch...

**Comment by fabiensc0ville** [comment]
A little follow up: disabling the _spell corrector_ greatly improves performances, particularly scrolling.

**Comment by HelperInCa** [comment]
> A little follow up: disabling the _spell corrector_ greatly improves performances, particularly scrolling.
Is this function *spell corrector* in Typora or just Mac? Could you please classify it? Maybe some screenshot will be better.

**Comment by fabiensc0ville** [comment]
Yes, I meant spelling and grammar from within the Typora app

**Comment by HelperInCa** [comment]
> Yes, I meant spelling and grammar from within the Typora app

@fabiensc0ville
Ok. I haven't use it ever. But there is still lag.

**Comment by raphaelhuleux** [comment]
In Typora 0.9.9.35.2, a big file with 100000 words and a lot of math is basically unusable, it takes ages to edit a math line, on a Macbook Pro 13' 2019.

**Comment by AlexFreik** [comment]
My Typora still too slow with a long dock with a lot of pictures inside so the best solution for me is divide files in 
00.md
01.md
...
66.md
and than use 
$  cat *.md > youroutputname.md
so I open long file only once at the final stage (to import in pdf, for example)

**Comment by Jemmy512** [comment]
Omg, this issue is still not fixed after 4 years later. Typora(4901)  is stuck when I'm scrolling on macOS Big Sur. Test file: https://github.com/Jemmy512/book-notes/blob/master/Linux/LinuxKernel.md

**Comment by leewi9** [comment]
5 years passed, a lot of things can change(e.g. President Trump come and go), but not this issue:)

**Comment by Fogest** [comment]
It's also the exact reason I haven't used this program in five years.

**Comment by Nighmared** [comment]
Typing inside math blocks gets incredibly laggy on linux in a file with 32k chars

**Comment by leowill01** [comment]
I am also suffering from extreme typing and scrolling lag on a doc with lots of images (~27k word/~120 image). I have other docs open in Typora at the same time and the lag goes away when I  switch to a smaller one, so this indicates it is a problem with handling large individual docs rather than a program- or system-wide problem.

- macOS Big Sur 11.1 (mid-2017 15" Macbook Pro)
- Typora 0.9.9.36.2 (4901)

**Comment by manasmbellani** [comment]
Same issue as @leowill01.

Also, noted that the issue is present in Typewriter Mode when Markdown is formatted. Issue is not present in Source Code Mode.

I have had to split a large document into smaller documents in folders to overcome this issue.

**Comment by leowill01** [comment]
I've noticed that since restarting my computer several times (just from normal usage—not just because of Typora), the performance in editing my large doc is noticeably improved since I last opened it with Typora. It does still tend to hang up sometimes, but nowhere near the perma-lag it was displaying before. Maybe an issue with how large docs are being handled in memory after being open for a while? Maybe an issue with memory clearance when Typora is quit and reopened later without a system reboot?

**Comment by morgenman** [comment]
Seems like the solution would be to cache the output, only evaluating sections that have changed. It would need to evaluate the whole doc when switching back from source code mode, as well as when doing something that effects more than the current line/block (like deleting an element from the beginning of a list). Given that it's just text processing, this doesn't seem like something that should be a problem. I can have a million pages open in a word processor with formatting and it doesn't lag.

**Comment by Fogest** [comment]
@morgenman It seems like this software is basically rendering the entire document even when only a small portion if it may be visible. That's what seems to be lagging it. 

So either only rendering what is visible plus a slight buffer above and below the current viewport should be enough to resolve this. Or a similar idea as mentioned above, the whole document is rarely changing so most of the markdown could have the output content cached so that it is not constantly re-generating the output.

**Comment by leowill01** [comment]
I can now confirm (for me at least) that the extreme lag starts immediately after I add a new image via drag-and-drop. 

A full reboot of my Mac followed by reopening the doc doesn't even repair the lag.

It seems that with every new image addition to my large doc, it has to wait an arbitrary amount of time for it to internally render the document or something? In the arbitrary meantime, it's essentially unusable.

**Comment by morgenman** [comment]
Yea. I just hit the doc size where I'm beginning to notice slowdowns. I got
sucked into the won't fix issue of making typora open source and they
reccomended marktext.
All of my typora files opened perfectly and idle cpu usage was down to
practically nothing, so I've switched.
So long and thanks for all the fish!

**Comment by manasmbellani** [comment]
I have just split my large file into smaller files inside a folder 

I turned on always show word count from the preferences. When I hit 6000 I start a new file.

I also use the search from left hand navigation in file view for searching as workaround.

**Comment by leowill01** [comment]
I now have noticed that it seems to not be the act of inserting a new image, but simply having an image within the current view in a large document (which includes my previous case of drag+drop inserting an image as it then becomes viewable). As soon as an image comes into the viewport, it begins to lag fiercely. When I scroll the image out of sight of the viewport, the lag disappears.

**Comment by gruvw** [comment]
I also have a lot of delay when typing (input lag) in a document loaded with math formulas (but only ~2200 words).
I am using the Linux version with an intel i5 and 16gb of ram in my pc...

**Comment by leowill01** [comment]
@gruvw Could you run a test: try and see if the lag disappears when no math formulas are in the viewport. Meaning, try and scroll to a place in your document (or make the app window smaller) to where only text is visible (no formulas, images, etc.) and see if the lag disappears. Then see if it reappears whenever you scroll a formula into view. I wonder if your issues with formulas is analogous to mine with images being in the viewport causing the lag.

**Comment by gruvw** [comment]
@leowill01 even without any formula being displayed on my screen, the input lag is still very present, my experience showed that it's when a document contains lots of $ and $$ elements that lag increases. It is not really affecting the scroll or loading time but more about the input lag when typing

**Comment by ArjanDeVries** [comment]
Just tried to copy my diary of 1.6MB with 40K lines in typora.  And scrolling in much slower then my ultraedit editor. But ok typora is not a text editor. But when i try to search something it becomes unresponsive. So too bad. This would be a nice alternative. Keep up the good work!

**Comment by tschrenk-corp** [comment]
I am seing the same issue. 
18310 words, 346 images of toal size 249 MB, on latest MacOS Big Sur 11.4.
I can easily reproduce. 

1. Create a blank typora file, and insert a picture, relative path. A subdirectory is created with the image in.
2. Copy the sample image until you have 300 images (mark all, CMD+C, CMD+v) duplicates
3. With Finder browse to the image directory, mark all with cmd+a, drag and drop them into the md file. It takes a while and the file shows the 300 images. Now try to insert text at the beginning.

The performance decrease is not happening simple after e number of graphics but it seems to be size related
with 346 images total size 249 MB it is not editable
with 512 objecs each 7 kb total 4,2 MB it is no problem at all

So I would think the rendering process needs to read through all images an that causes the slow performance when the image size in total gets higher. I sent a sample via mail.

**Comment by tschrenk-corp** [comment]
Also tried it with big images 65 images with total 271 MB works ok
with 128 files and 543 MB ist gets slow.

So we have
- 249 MB 346 images slow
+ 4,2 MB 512 images fast
+ 8,4 MB, 1024 images still ok 
+ 271 MB 64 images ok
- 543 MB 128 images slow

So I would say with raising number of files the app gets slower
If files are bigger it gets slower more quickly.
So it is a combination of number of files and total size.

**Comment by leowill01** [comment]
I'm not sure what changed but I am no longer experiencing this issue. If I remember correctly, even after updating to the most recent version (`Version 0.10.11 (5372)`), I was still initially experiencing the issue, but it seems it has since disappeared. So, something changed and I'm no longer experiencing lag when scrolling through a single ~160 image (~46MB), ~34K word document.

**Comment by chrisoffner** [comment]
> @leowill01 even without any formula being displayed on my screen, the input lag is still very present, my experience showed that it's when a document contains lots of $ and $$ elements that lag increases. It is not really affecting the scroll or loading time but more about the input lag when typing
> 
> Performance issues (input lag) should be Typora number 1 priority.

Same for me, using Typora 0.11.2 (5601) on macOS Big Sur 11.4 (20F71).
I use Typora for study notes at university, including code sections (```) as well as LaTeX/MathJax ($$) and after a certain size editing it just slows to an unusable crawl, even with just a single small SVG file included.

**Comment by Jalr4ever** [comment]
2021/8/4, this issue still hasn't be solved ... In my scenarios it's mostly large text content with fewer images. At last, I come to use VS Code...

**Comment by Jalr4ever** [comment]
2021/8/14, Good news，seems like typora improve its performance in 0.11.4 version. I tested with a few images and 60k words, well it's not lagging anymore, have a try!

**Comment by n1vk** [comment]
This problem is still unfixed. I've got a Markdown file with about only 3000 words, but mostly **Chinese characters and Math formulas**. And now Chinese typing became laggy, and it feels painful when typing a lot of content.

In Typora:
![ezgif-6-c18adc24022e](https://user-images.githubusercontent.com/73031052/139570932-3d8cd376-62e3-41f3-88b6-602cca87129e.gif)

In system memo:
![ezgif-6-979366f44941](https://user-images.githubusercontent.com/73031052/139570999-50f02141-0d6e-4535-a138-4ffef071decd.gif)


Device: MacBook Pro 15-inch, 2019
OS: macOS Monterey 12.0.1
Typora Version: 0.11.5 (5648)
RAM: 16GB

**Comment by Denly** [comment]
Hi, this is a paid user who really loves Typora, but for larger files, it is so slow to a point that can't be used. My file is mixed with English, Image, Mandarin. I tried other Markdown editors that don't have this issue. Hope this can be resolved.

**Comment by arthurjosse** [comment]
Generally slow, and annoying launch time. Opening a note should be instant. Was waiting for performance improvement, but I won't use Typora anymore due to lack of speed.

**Comment by morgenman** [comment]
I will say for anyone still frustrated by this issue that ObsidianMD now has live preview which is very similar to how typora performs out of the box. 
Also, plugins... so many plugins for basically anything you could want it to do.

**Comment by Denly** [comment]
I have moved to [Obsidian](https://obsidian.md/) too. Obsidian's live preview mode is the same as Typora, and it's free with iCloud sync, with no performance issue so far. I really hope Typora can solve this one day anyway.

**Comment by pepsilon7** [comment]
I have the same problem and I really hope that this issue gets fixed soon, because I really like Typora it is pretty much what I could have ever asked for and I am really happy with it. However, if this issue will not get resolved changing to another editor might be the only remaining option.
Also it would be great to hear if there is any progress on this issue.

**Comment by Fogest** [comment]
> I have the same problem and I really hope that this issue gets fixed soon, because I really like Typora it is pretty much what I could have ever asked for and I am really happy with it. However, if this issue will not get resolved changing to another editor might be the only remaining option. Also it would be great to hear if there is any progress on this issue.

This has been a problem present for the last 6 years and there has been no activity on this issue despite the popularity. I would suggest just moving to a different editor, this one is shit.

**Comment by pepsilon7** [comment]
> > I have the same problem and I really hope that this issue gets fixed soon, because I really like Typora it is pretty much what I could have ever asked for and I am really happy with it. However, if this issue will not get resolved changing to another editor might be the only remaining option. Also it would be great to hear if there is any progress on this issue.
> 
> This has been a problem present for the last 6 years and there has been no activity on this issue despite the popularity. I would suggest just moving to a different editor, this one is shit.

Yes I guess this will be the only option then. However I really like the image paste functionality of typora and have not seen that in any other editors.
Is there a way to achive the same functionality in Obsidian?

**Comment by ley7952** [comment]
Hi I have a similar issue here. Once my markdown file gets larger than ~10Kb (I tested this by copy-pasting an existing laggy markdown file into a new blank one paragraph by paragraph), the cursor movement is visibly slower. Say I want to move the cursor to 10 characters away, I have to stroke my keyboard 10 times instead of pressing → and release when the cursor is in the right position (because the cursor does not move at all if I don't realse my finger). I don't find this issue on other editors or on other smaller files.

**Comment by javacaoyu** [comment]
Hi, this is a paid user who really loves Typora.
I hope that performance will be fixed because i so like typora.
by the way , if not fix may i change to other software even im paid in typora.
but im so like it, i don't want to change to other.

Thanks for typora developer, hope you guys can fixed that.
Despite the performance issues, I am still very grateful for the help from typora.

**Comment by tgeijten** [comment]
I'm seeing severe performance issues ever since Typora 1.2.0, which have not been present before. Here is a link to one of the documents I'm editing that has this issue:

https://github.com/hyfydy/hyfydy-doc/blob/main/hyfydy-user-manual.md

Move the cursor to the text under "About" and hold the right cursor key to see the problem. Downgrading to Typora 1.1.5 resolves the issue completely.

Also, note that the issue only occurs when navigating normal text. The cursor moves just fine inside code blocks or equations.

**Comment by jesuslop** [comment]
Paid guy also. Just confirm the performance issues, specially bad with code and math sections. Updated Typora on win, modern computer, 20K words document (and math and code). I type stuff and have time to sigh waiting for the effects to happen. Several seconds response time. Big lag entering/leaving code/math sections. Hope you spot root cause, for Typora has a concept and features unsustitutable.

**Comment by tgeijten** [comment]
For me, all performance issues have been resolved in 1.4.4. Thank you!

**Comment by lizheng19581010** [comment]
still laggy in 1.4.4  : (

**Comment by PRO-2684** [comment]
In my case I'll have to wait for a few seconds when opening a file of `20k` size, but the editing is not laggy after having opened it.

Windows, Typora 1.5.12

**Comment by rkingery** [comment]
Also having this issue. It seems to be directly related to using a lot of math expressions. When documents full of math get longer than say 2-3 pages things slow down considerably. I've found no way to fix this other than force quit and relaunch, which only sometimes works, and only for a few minutes.

**Comment by zhwwng** [comment]
Long time Typora user here. Haven't faced this issue until have to write some math-heavy documents recently. The laggy input is pretty frustrating. Hope it can be fixed soon.

**Comment by YYACMU** [comment]
> Long time Typora user here. Haven't faced this issue until have to write some math-heavy documents recently. The laggy input is pretty frustrating. Hope it can be fixed soon.

me too, light use is perfect, but heavy use is not good. I have to open a new temp file for subsections to get a quicker experience.

**Comment by ShijianXu** [comment]
When my document contains too many (hundreds of) mathematical formulas, Typora becomes very laggy and difficult to operate. This is becoming even unbearable. Hope it can be fixed soon.

BTW, is there any other light-weighted editors like Typora, apart from [Obsidian](https://obsidian.md/)?

**Comment by bestcoder** [comment]
Also noticing 5-10 second pauses while typing in a 125k file. It seems to worsen when there are more image references. I also have a lot of url references but don't whether that has any impact. The image references do for sure. Running version 1.8.10.

**Comment by famfigueiredo** [comment]
Long time (paid) Typora user on MacOS. Currently trying to work with a 35k file, but it seems impossible. It won't even load past the first page.

**Comment by raphaelhuleux** [comment]
Same here. Obsidian, with the same live preview features, works perfectly fine.On 12 Jun 2024, at 13:19, Filipe Figueiredo ***@***.***> wrote:﻿
Long time (paid) Typora user on MacOS. Currently trying to work with a 35k file, but it seems impossible. It won't even load past the first page.

**Comment by shoesforindustry** [comment]
Same here on Mac M3 / 24GB / MacOS 14.5 (23F79), Typora Version 1.9.4 (7260). Document is markdown only, 257KB.
Unusable to edit this document, huge lag, very slow. This is not a slow machine and I have no trouble with other markdown editors.

**Comment by AhmedThahir** [comment]
Any updates?

**Comment by famfigueiredo** [comment]
> Any updates?

It definitely runs much smoother for me, and I am working with massive files with a bunch of plots and very large tables.

**Comment by wrendx** [comment]
I have the same problem, Has anyone solved it?

Device: MacBook Pro 15-inch, 2019
OS: macOS Monterey 12.6.6
Typora Version: 1.9 
RAM: 16GB

**Comment by famfigueiredo** [comment]
> I have the same problem, Has anyone solved it?
> 
> Device: MacBook Pro 15-inch, 2019 OS: macOS Monterey 12.6.6 Typora Version: 1.9 RAM: 16GB


> It definitely runs much smoother for me, and I am working with massive files with a bunch of plots and very large tables.

Apple M2 Pro, 16 GB RAM, 2023
OS: Sonoma 15.0.1
Typora: Version 1.9.4 (7260)

**Comment by fromXin** [comment]
Have the same problem

Windows 11 Pro for Workstations

Typora: 1.9.5

It has been 8 years since 2016.

**Comment by JYbill** [comment]
i have compare with v1.6.7 and v1.7.0+, found v1.6.7 performance is fast, others is slow.
The interaction will be slow but acceptable in v1.6.7 When the document is 65,000 words long and edited

Devices
- Apple M1
  - 15.0.1
- Apple Intel(i5)
  - sonoma 14.5

**Comment by robb-brown** [comment]
I have a document that's only a couple pages long but contains a lot of equations. With the latest version (1.11.7) it is noticeably slow while typing. As suggested by @JYbill I reinstalled 1.6.7 and the delay when typing is not there.

Would be great to get this fixed before adding more features.
